plugin/db/mysql/dump.go

```diff
@@ -41,58 +34,60 @@ const (
 		"-- View structure for `%s`\n" +
 		"--\n" +
 		"%s;\n"
-	routineStmtFmt = "" +
-		"--\n" +
-		"-- %s structure for `%s`\n" +
-		"--\n" +
-		settingsStmt +
-		"DELIMITER ;;\n" +
-		"%s ;;\n" +
-		"DELIMITER ;\n"
-	eventStmtFmt = "" +
-		"--\n" +
-		"-- Event structure for `%s`\n" +
-		"--\n" +
-		settingsStmt +
-		"SET time_zone = '%s';\n" +
-		"DELIMITER ;;\n" +
-		"%s ;;\n" +
-		"DELIMITER ;\n"
-	triggerStmtFmt = "" +
-		"--\n" +
-		"-- Trigger structure for `%s`\n" +
-		"--\n" +
-		settingsStmt +
-		"DELIMITER ;;\n" +
-		"%s ;;\n" +
-		"DELIMITER ;\n"
-)
-
-var (
-	excludeAutoIncrement = regexp.MustCompile(`AUTO_INCREMENT=\d+ `)
 )
 
 // Dump dumps the database.
 func (driver *Driver) Dump(ctx context.Context, database string, out io.Writer, schemaOnly bool) (string, error) {
 	// mysqldump -u root --databases dbName --no-data --routines --events --triggers --compact
 
+	var err error
 	// We must use the same MySQL connection to lock and unlock tables.
 	conn, err := driver.db.Conn(ctx)
 	if err != nil {
 		return "", err
 	}
 	defer conn.Close()
 
+	mysqldumpArgs := []string{
```

Comment:
Please take a look at `--single-transaction` and `--source-data=2`, we need them for the same snapshot behavior.

Reply:
After this change, we can parse the binlog info from backup file like:
`CHANGE MASTER TO MASTER_LOG_FILE='binlog.000099', MASTER_LOG_POS=157;`

Reply:
Based on offline chat, we lock tables and unlock tables by ourselves instead of using `--single-transaction`. More, we get the master status by ourselves, too.